respa_admin/views/resources.py

```diff
@@ -86,6 +89,66 @@ def get_queryset(self):
         return qs
 
 
+class ManageUserPermissionsView(ExtraContextMixin, UpdateView):
+    model = User
+    context_object_name = 'user_object'
+    pk_url_kwarg = 'user_id'
+    form_class = UserForm
+    template_name = 'respa_admin/resources/edit_user.html'
+
+    def get_success_url(self, **kwargs):
+        return reverse_lazy('respa_admin:edit-user', kwargs={'user_id': self.object.pk})
+
+    def _validate_forms(self, form, unit_authorization_formset):
+        valid_form = form.is_valid()
+        unit_authorization_formset = unit_authorization_formset.is_valid()
+
+        return valid_form and unit_authorization_formset
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        context['unit_authorization_formset'] = get_unit_authorization_formset(
+            self.request,
+            instance=self.object,
+        )
+        return context
+
+    def post(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        form = self.get_form()
+
+        unit_authorization_formset = get_unit_authorization_formset(request=request, instance=self.get_object())
+
+        if self._validate_forms(form, unit_authorization_formset):
+            return self.forms_valid(form, unit_authorization_formset)
+        else:
+            return self.forms_invalid(form, unit_authorization_formset)
+
+    def forms_valid(self, form, unit_authorization_formset):
+        self.object = form.save()
+        unit_authorization_formset.instance = self.object
+        for formset in unit_authorization_formset.cleaned_data:
```

Comment:
I guess we are iterating through forms, not formsets here. Variable `formset` is a bit misleading.